File: 26. 删除有序数组中的重复项/main.rs
struct Solution {

}

impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        if nums.len() < 2 {
            return nums.len() as i32
        }

        let mut left: usize = 0;
        let mut right: usize = 1;
        while right < nums.len() {
            if nums[left] == nums[right] {
                right += 1;
            } else {
                nums[left + 1] = nums[right];
                left += 1;
                right += 1;
            }
        }
        return left as i32 + 1;
    }
}

fn main() {
    let mut list = vec![1,1,2];
    let result = Solution::remove_duplicates(&mut list);
    println!("{:?}", result);
}

File: 26. 删除有序数组中的重复项/swift/main/main/main.swift
//
//  main.swift
//  main
//
//  Created by cedricwu on 4/24/25.
//

import Foundation

class Solution {
    func removeDuplicates(_ nums: inout [Int]) -> Int {
        guard nums.count > 1 else {
            return 1
        }
        
        var left = 0
        var right = 1
        
        while right < nums.count {
            if nums[left] >= nums[right] {
                right += 1
            } else {
                left += 1
                swap(&nums, i: left, j: right)
            }
        }
        
        return left + 1
    }
    
    func swap(_ nums: inout [Int], i: Int, j: Int) {
        let tmp = nums[i]
        nums[i] = nums[j]
        nums[j] = tmp
    }
}

let s = Solution()
var list = [0,0,1,1,1,2,2,3,3,4]
print(s.removeDuplicates(&list))
print(list)
